get-extracts/get-extracts.py

```diff
@@ -1,146 +1,81 @@
-
-        #set the extract type by according to the file name prefix
-        if '_20' in x['name']:
-            sep = x['name'].index('_20')
-            ftype = x['name'][:sep]
-
-            if ftype in files:
-                #if this data set already exist, check if this file is newer.
-                if x['mtime'] > files[ftype]['age']:
-                    files[ftype]['name'] = x['name']
-                    files[ftype]['url'] = x['url']
-                    files[ftype]['age'] = x['mtime']
-            else:
-                #If no other files of this data set exist, and this file.
-                files[ftype] = {}
-                files[ftype]['name'] = x['name']
-                files[ftype]['url'] = x['url']
-                files[ftype]['age'] = x['mtime']
-
-    #loop through sub directories
-    if x['type'] == 'dir':
-        subdir = theurl + x['url']
-
-        subresult = urllib2.urlopen(subdir)
-        subjson = json.loads(subresult.read())
-
-        for y in subjson:
-            #exclude directories and column header files
-            if y['type'] == 'file' and y['name'].find('headers') == -1:
-
-                #set the extract type by according to the file name prefix
-                if '_20' in y['name']:
-                    sep = y['name'].index('_20')
-                    ftype = y['name'][:sep]
-
-                    if ftype in files:
-                        #if this data set already exist, check if this file is newer.
-                        if y['mtime'] > files[ftype]['age']:
-                            files[ftype]['name'] = y['name']
-                            files[ftype]['url'] = y['url']
-                            files[ftype]['age'] = y['mtime']
-                    else:
-                        #If no other files of this data set exist, and this file.
-                        files[ftype] = {}
-                        files[ftype]['name'] = y['name']
-                        files[ftype]['url'] = y['url']
-                        files[ftype]['age'] = y['mtime']
-
-
-if len(files) == 0:
-    print "No data extract files found. If this is an error, please contact your technical account manager."
-
-#download most recent files
-for key, value in files.iteritems():
-    fname = value['name']
-    furl = value['url']
-    u = urllib2.urlopen(furl)
-    flocation = storageDir + '/' + fname
-    f = open(flocation, 'wb')
-    meta = u.info()
-    file_size = int(meta.getheaders("Content-Length")[0])
-    print "Downloading: %s Bytes: %s" % (fname, file_size)
-
-    file_size_dl = 0
-    block_sz = 8192
-    while True:
-        buffer = u.read(block_sz)
-        if not buffer:
-            break
-
-        file_size_dl += len(buffer)
-        f.write(buffer)
-        status = r"%10d  [%3.2f%%]" % (file_size_dl, file_size_dl * 100. / file_size)
-        status = status + chr(8)*(len(status)+1)
-        print status,
-
+sort_files_and_directories(content)
```

Comment:
I find it pretty odd that `sort_files_and_directories` is modifying this global `files` variable... it would 💥 if that was not defined.
How about we let that function own the `files` object and just do this here:
files = sort_files_and_directories(content)